### PR TITLE
Fixed help command

### PR DESCRIPTION
**Pull request by AndrewJWin**
They are as such:

-  Fixed filtering for certain categories, main one being `social`.
-  Added Owner only override for NSFW command listing.
-  Fixed miscellaneous issues that occurred while doing prior fixes.

### CODE REVIEW

File: commands/general/help.js [hunk 1]
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
...
         .addField("Categories", output);
+
     } else {
-      let n = 0;
-      sorted.forEach(c => {
-        if (c.help.category.toLowerCase() === type.toLowerCase()) {
-          n++;
-        }
-      });
+      if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
```
Comment (MrJacz) [inline commands/general/help.js]: you do know that `this.client.aliases` is a thing?

File: commands/general/help.js [hunk 2]
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
...
+        const cm = this.client.commands.get(type) || this.client.commands.get(this.client.aliases.get(type));
+        if (level < this.client.levelCache[cm.conf.permLevel]) return;
+        embed.setTitle(cm.help.name.toProperCase())
+          .addField("Command description", cm.help.description)
+          .addField("Command usage", `\`${cm.help.usage}\``)
+          .addField("Command aliases", cm.conf.aliases.length == 0 ? "None" : cm.conf.aliases.join(", ") );
```
Comment (MrJacz) [inline commands/general/help.js]: use `===`

File: commands/general/help.js [hunk 3]
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
...
+
+      } else {
+        let n = 0;
+        sorted.forEach(c => {
+          if (c.help.category.toLowerCase() === type) {
```
Comment (MrJacz) [inline commands/general/help.js]: this can be a one liner

File: commands/general/help.js [hunk 4]
```diff
@@ -32,50 +33,47 @@ class Help extends Command {
...
-      for (const c of sorted.values()) {
-        if (c.help.category.toLowerCase() === type.toLowerCase()) {
-          if (c.help.category === "Owner" && level < 10 ) return;
-          if (c.help.category === "NSFW" && !message.channel.nsfw) return;
-          if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
-            if (level < this.client.levelCache[c.conf.permLevel]) return;
-            output += `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 50 ? c.help.description.slice(0,50) +"...": c.help.description}`;
+        let output = "";
+        let num = 0;
+        const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
```
Comment (MrJacz) [inline commands/general/help.js]: parse the number once instead of doing it multiple time

**Review by MrJacz — CHANGES_REQUESTED** [review]

**Comment by AndrewJWin** [comment]
-  Fixed L40-C69 to check against aliases.
-  Fixed L46-C61 to strict match case.
-  Fixed L50-L51 to consolidate into one line.
-  Fixed multiple parseInt(page) to a single parseInt() at the beginning of the code.

**Comment by AndrewJWin** [comment]
Included is the a new command: **Vaportype**.

-  Spaces out a given word/sentence.
-  Automagically changes to uppercase.
-  Looks super cool.

**Comment by YorkAARGH** [comment]
Can you make the vaportype command a separate PR please?

File: commands/fun/vaportype.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      name: "vaportype",
+      description: "W O A H  D U D E  S P A C E D  O U T",
+      usage: "vaportype <text>",
+      category: "fun",
+      aliases: ["vapor", "vpt"]
+    });
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
+    args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: You should be checking args then returning that message instead

File: commands/fun/vaportype.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  async space(message, args) {
+    let msg = "";
+    await args.forEach(c => msg += c.toUpperCase().split("").join(" ") + "  ");
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: forEach returns void which makes it un-`await`able

File: commands/fun/vaportype.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    message.channel.send(msg);
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: return

File: commands/fun/vaportype.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+      category: "fun",
+      aliases: ["vapor", "vpt"]
+    });
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
+    args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");
+  }
+
+  async space(message, args) {
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: Also, this function really doesn't need to be a thing, considering few things it does.

File: commands/fun/vaportype.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    super(client, {
+      name: "vaportype",
+      description: "W O A H  D U D E  S P A C E D  O U T",
+      usage: "vaportype <text>",
+      category: "fun",
+      aliases: ["vapor", "vpt"]
+    });
+  }
+
+  async run(message, args, level) { // eslint-disable-line no-unused-vars
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: this can be `async run(message, args) {`

**Review by MrJacz — CHANGES_REQUESTED** [review]
The simple use of a for loop would be the suggested way to add the spaces...

File: commands/general/help.js [hunk 10]
```diff
@@ -16,13 +16,17 @@ class Help extends Command {
 
   async run(message, [type, page], level) {
 
+    type ? type = type.toLowerCase() : undefined;
```
Comment (MrJacz) [inline commands/general/help.js]: Assigning with a ternary is a bad practice so `type = type ? type.toLowerCase() : null`
Use `null` instead (More style than anything tho)

File: commands/general/help.js [hunk 11]
```diff
@@ -16,13 +16,17 @@ class Help extends Command {
...
+    page ? page = parseInt(page) : undefined;
```
Comment (MrJacz) [inline commands/general/help.js]: Same here

File: commands/general/help.js [hunk 12]
```diff
@@ -32,52 +36,47 @@ class Help extends Command {
...
         .addField("Categories", output);
+
     } else {
-      let n = 0;
-      sorted.forEach(c => {
-        if (c.help.category.toLowerCase() === type.toLowerCase()) {
-          n++;
-        }
-      });
+      if (this.client.commands.has(type) || this.client.aliases.has(type)) {
```
Comment (MrJacz) [inline commands/general/help.js:43]: A simple if no then return check would be better in this case

**Review by MrJacz — CHANGES_REQUESTED** [review]

File: commands/fun/vaportype.js (hunk 5 above)
Comment (AndrewJWin) [reply commands/fun/vaportype.js]: I am, with ``args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");`` it verifies ``args[0]`` with any value provided, which means if you don't have anything, i.e. ``m.vaportype``, it'll respond with the ``"you have to provide me text to space out you baka."``.

File: commands/general/help.js (hunk 10 above)
Comment (AndrewJWin) [reply commands/general/help.js]: No, because it needs to be ``undefined``, not ``null``.

File: commands/general/help.js (hunk 11 above)
Comment (AndrewJWin) [reply commands/general/help.js]: Same with above.

File: commands/general/help.js (hunk 12 above)
Comment (AndrewJWin) [reply commands/general/help.js:43]: This is just fixing the currently uploaded version, no need for major changes.

File: commands/fun/vaportype.js (hunk 8 above)
Comment (AndrewJWin) [reply commands/fun/vaportype.js]: I wanted to get practice with class functions.

File: commands/fun/vaportype.js (hunk 9 above)
Comment (AndrewJWin) [reply commands/fun/vaportype.js]: Wasn't sure if that would work with how we execute commands, went ahead and did it anyway.

File: commands/general/help.js (hunk 10 above)
Comment (kyranet) [reply commands/general/help.js]: @MrJacz is right.

```javascript
if (type) type = type.toLowerCase();
```

File: commands/general/help.js (hunk 12 above)
Comment (MrJacz) [reply commands/general/help.js]: This wouldnt be major.

File: commands/fun/vaportype.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    });
+  }
+
+  async run(message, args) {
+    args[0] ? this.space(message, args) : message.reply("you have to provide me text to space out you baka.");
+  }
+
+  async space(message, args) {
+    let msg = "";
+    for (const c of args) { msg += c.toUpperCase().split("").join(" ") + "  "; }
```
Comment (MrJacz) [inline commands/fun/vaportype.js]: The brackets are useless

**Review by MrJacz — CHANGES_REQUESTED** [review]

**Review by MrJacz — APPROVED** [review]

**Pull request merged**